modules/telestion-application/src/main/java/de/wuespace/telestion/application/loader/Loader.java

```diff
@@ -0,0 +1,350 @@
+
+import de.wuespace.telestion.api.DefaultConfigurable;
+import de.wuespace.telestion.application.deployment.Deployment;
+import de.wuespace.telestion.application.launcher.ConfigDeploymentLauncher;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A loader is a collection of event handlers that hook into different processes of the associated
+ * {@link ConfigDeploymentLauncher Launcher}.
+ * <p>
+ * It can register event handlers in:
+ * <ul>
+ *     <li>the startup and shutdown process of the loader</li>
+ *     <li>the {@link io.vertx.core.Verticle Verticle} deployment and un-deployment</li>
+ *     <li>the Vertx exception handling</li>
+ * </ul>
+ * <p>
+ * To start, take a look at the {@link TelestionLoader} as an abstract reference implementation.
+ *
+ * @param <T> the type of the loader configuration
+ * @author Ludwig Richter (@fussel178)
+ * @see TelestionLoader
+ * @see LoaderConfiguration
+ * @see ConfigDeploymentLauncher
+ */
+public interface Loader<T extends LoaderConfiguration> extends DefaultConfigurable<T> {
+
+	/**
+	 * Instantiates a new loader from a given classname.
+	 *
+	 * @param className the classname or binary name of the loader class
+	 * @return the loader instance associated to the classname
+	 */
+	static Loader<?> instantiate(String className)
+			throws ClassNotFoundException, InvocationTargetException, InstantiationException, IllegalAccessException,
+			NoSuchMethodException, ClassCastException {
+
+		try {
+			var classLoader = Loader.class.getClassLoader();
+			var loaderClass = classLoader.loadClass(className);
+			return (Loader<?>) loaderClass.getConstructor().newInstance();
+		} catch (ClassNotFoundException e) {
+			throw new ClassNotFoundException(("Cannot find loader with classname %s. Please check the given " +
+					"classname, install required packages and try again.").formatted(className), e.getException());
+		} catch (InvocationTargetException e) {
+			throw new InvocationTargetException(e.getTargetException(), ("The loader %s cannot be created because " +
+					"the constructor throws an exception during instantiation. Please check your loaders " +
+					"constructor, fix the breaking code and try again.").formatted(className));
+		} catch (InstantiationException e) {
+			throw new InstantiationException(("Abstract loaders aren't supported. Please extend the \"Loader\" " +
+					"class in loader %s, add the sub class to the configuration and try again.").formatted(className));
+		} catch (IllegalAccessException e) {
+			throw new IllegalAccessException(("The class loader cannot instantiate the loader %s because " +
+					"the constructor is inaccessible. Please add at least one public constructor with no arguments " +
+					"to continue.").formatted(className));
```

Comment:
Can there be more than one constructor with no arguments?

```suggestion
					"the constructor is inaccessible. Please add a public constructor with no arguments " +
					"to continue.").formatted(className));
```

Reply:
Good point.